sdk/spring/scripts/replace_util.py

```diff
@@ -0,0 +1,178 @@
+    )
+    parser.add_argument(
+        '--log',
+        type = str,
+        choices = ['debug', 'info', 'warn', 'error', 'none'],
+        required = False,
+        default = 'info',
+        help = 'Set log level.'
+    )
+    parser.add_argument(
+        '--color',
+        type = str,
+        choices = ['true', 'false'],
+        required = False,
+        default = 'true',
+        help = 'Whether need colorful log.'
+    )
+    return parser.parse_args()
+
+
+def init_log(args):
+    log_dict = {
+        'debug': Log.DEBUG,
+        'info': Log.INFO,
+        'warn': Log.WARN,
+        'error': Log.ERROR,
+        'none': Log.NONE
+    }
+    log.set_log_level(log_dict[args.log])
+    color_dict = {
+        'true': True,
+        'false': False
+    }
+    log.set_color(color_dict[args.color])
+
+
+def replace(module):
+    """
+    Replace action
+    :param module:
+    :return:
+    """
+    module = 'spring'
+    for pom in config[module].keys():
+        replace_artifact_id(module, pom)
+        replace_version(module, pom)
+
+
+def replace_artifact_id(module, pom):
+    """
+    Replace artifactId in dependency
+    :param module module name
+    :param pom: pom file path
+    :return:
+    """
+
+    log.info('Replacing artifact id in file: {}'.format(pom, module))
+    artifact_id_tuple = config[module][pom]['artifact_id_tuple']
+    log.info('Module: {}, artifact ids: {}'.format(module, artifact_id_tuple))
+    with in_place.InPlace(pom) as file:
+        line_num = 0
+        for line in file:
+            line_num = line_num + 1
+            for artifact_id_mapping_tuple in artifact_id_tuple:
+                new_line = line.replace(artifact_id_mapping_tuple[0], artifact_id_mapping_tuple[1])
+                if line != new_line:
+                    log.debug('Updated artifact id line {}'.format(line_num))
+                    log.debug('    old_line = {}.'.format(line.strip('\n')))
+                    log.debug('    new_line = {}.'.format(new_line.strip('\n')))
+                    line = new_line
+            file.write(line)
+
+
+def replace_version(module, pom):
+    """
+    Replace version, contains dependency and plugin parts.
+    :param module module name
+    :param pom: pom file path
+    :return:
+    """
+
+    log.info('Replacing version in file: {}'.format(pom))
+    version_tuple = config[module][pom]['version_tuple']
+    log.info('Module: {}, versions: {}'.format(module, version_tuple))
+    with in_place.InPlace(pom) as file:
+        line_num = 0
+        for line in file:
+            line_num = line_num + 1
+            for version_mapping_tuple in version_tuple:
+                dependency_update_flag = '<!-- {x-version-update;' + version_mapping_tuple[0] + ';'
+                plugin_update_flag = '<!-- {x-include-update;' + version_mapping_tuple[0] + ';'
+                # update version in dependency part
+                if dependency_update_flag in line:
```

Comment:
How about do like this:

Define global variable:

X_VERSION_UPDATE = 'x-version-update'
X_INCOUDE_UPDATE = 'x-include-update'.

Then the condition check will be like this:
```
if (X_VERSION_UPDATE in line or X_VERSION_UPDATE in line) and version_update_item.id in line:
```

Reply:
If we add check logic above, we do not need check `X_VERSION_UPDATE` and `X_INCLUDE_UPDATE ` here any more.